### DISCUSSION TITLE
Is there any documentation for the python api?

### DISCUSSION THREAD

**Discussion by Errorbot1122** [liquidctl/liquidctl#552] [General]
It would be useful to know if there was more documentation for the API then just [this](https://github.com/liquidctl/liquidctl#using-liquidctl-in-other-programs-and-scripts).

**Comment by jonasmalacofilho** [comment]
The public API has relatively good docstring coverage. Check the sources or use pydoc:

```
$ pydoc liquidctl.find_liquidctl_devices
```

All drivers implement `liquidctl.driver.base.BaseDriver` (although some methods may raise `NotSupportedByDevice` or `NotSupportedByDriver`), so checking the `BaseDriver` docs should give you a general idea of what can be done:

```
$ pydoc liquidctl.driver.base.BaseDriver
```

In some cases the documentation for the specific driver/method you're interested in can include additional information:

```
$ pydoc liquidctl.driver.commander_pro.CommanderPro.set_color
```

**Reply by Errorbot1122** [reply]
when ever it try to use `pydoc` in the local directory, I get this error:

```
problem in liquidctl - ModuleNotFoundError: No module named 'usb'
```

I've done everything i can:

 - Install PyUSB using pip
 - Installing LibUSB also using pip
 - Installing liquidctl inside the repo directory
 - Etc...

The only thing I have done is installing the **DLL** as I don't understand how to do currently and I'm worried it might brake my computer...

**Reply by jonasmalacofilho** [reply]
My first guess is that you have multiple Python installations and/or environments going on, and things are getting installed or called with different contexts.

Try the following, on a fresh _and activated_ virtual environment:

```
python --version
python -m pip install liquidctl
python -m pydoc liquidctl.driver.commander_pro.CommanderPro.set_color
```

**Comment by henryruhs** [comment]
We could at least throw the pydoc output into a gitbook.com page or similar.

**Reply by jonasmalacofilho** [reply]
It would be nice to have a GitHub workflow to publish the API documentation (and, later, the rest of the docs) to a website.
